assertpy/assertpy.py

```diff
@@ -156,948 +135,10 @@ def __init__(self, val, description='', kind=None, expected=None):
-            else:
-                err_msg = 'Expected attribute <%s>, but val has no attribute <%s>.' % (attr_name, attr_name)
-
-        def _wrapper(*args, **kwargs):
-            if err_msg:
-                self._err(err_msg) # ok to raise AssertionError now that we are inside wrapper
-            else:
-                if len(args) != 1:
-                    raise TypeError('assertion <%s()> takes exactly 1 argument (%d given)' % (attr, len(args)))
-
-                try:
-                    val_attr = getattr(self.val, attr_name)
-                except AttributeError:
-                    val_attr = self.val[attr_name]
-
-                if callable(val_attr):
-                    try:
-                        actual = val_attr()
-                    except TypeError:
-                        raise TypeError('val does not have zero-arg method <%s()>' % attr_name)
-                else:
-                    actual = val_attr
-
-                expected = args[0]
-                if actual != expected:
-                    self._err('Expected <%s> to be equal to <%s> on %s <%s>, but was not.' % (actual, expected, 'key' if is_dict else 'attribute', attr_name))
-            return self
-
-        return _wrapper
-
-### expected exceptions ###
-    def raises(self, ex):
-        """Asserts that val is callable and that when called raises the given error."""
-        if not callable(self.val):
-            raise TypeError('val must be callable')
-        if not issubclass(ex, BaseException):
-            raise TypeError('given arg must be exception')
-        return AssertionBuilder(self.val, self.description, self.kind, ex)
-
-    def when_called_with(self, *some_args, **some_kwargs):
-        """Asserts the val callable when invoked with the given args and kwargs raises the expected exception."""
-        if not self.expected:
-            raise TypeError('expected exception not set, raises() must be called first')
-        try:
-            self.val(*some_args, **some_kwargs)
-        except BaseException as e:
-            if issubclass(type(e), self.expected):
-                # chain on with exception message as val
-                return AssertionBuilder(str(e), self.description, self.kind)
-            else:
-                # got exception, but wrong type, so raise
-                self._err('Expected <%s> to raise <%s> when called with (%s), but raised <%s>.' % (
-                    self.val.__name__,
-                    self.expected.__name__,
-                    self._fmt_args_kwargs(*some_args, **some_kwargs),
-                    type(e).__name__))
-
-        # didn't fail as expected, so raise
-        self._err('Expected <%s> to raise <%s> when called with (%s).' % (
-            self.val.__name__,
-            self.expected.__name__,
-            self._fmt_args_kwargs(*some_args, **some_kwargs)))
-
-### helpers ###
     def _err(self, msg):
```

Comment:
`_err` is defined here, which uses `print`, so does assertpy.py need to do `from __future__ import print_function` opposed to the other files?

Reply:
correct, but per #97 probably replace `print(out)` with `logger.warning(out)`